app/services/rbac_api.rb

```diff
@@ -1,32 +1,63 @@
 # frozen_string_literal: true
 
-# This class is meant to handle calls to the RBAC light API
+require 'insights-rbac-api-client'
+
+# This class is meant to handle calls to the RBAC API
 class RbacApi
+  COMPLIANCE_FULL_ACCESS = 'compliance:*:*'
+  APPS = 'compliance,inventory'
+  ANY = '*'
+
   def initialize(b64_identity)
-    @url = URI.parse("#{URI.parse(Settings.rbac_url)}"\
-                     "#{Settings.path_prefix}/rbac/v1/access/")
+    @client = RBACApiClient::AccessApi.new
     @b64_identity = b64_identity
   end
 
-  def check_user
+  # rubocop:disable Metrics/MethodLength
+  def load_users_permissions
+    opts = {
+      limit: 10,
+      offset: 0
+    }
     begin
-      body = JSON.parse(access_check_response.body)
-      body['data'].find do |access_check|
-        return true if access_check['permission'] == 'compliance:*:*'
-      end
-    rescue Faraday::ClientError => e
-      Rails.logger.info("#{e.message} #{e.response}")
+      result = @client.get_principal_access(
+        RbacApi::APPS,
+        opts.merge({ header_params: { 'X_RH_IDENTITY': @b64_identity } })
+      )
+      result.data
+    rescue RBACApiClient::ApiError => e
+      Rails.logger.info(e.message)
+      # TODO: unauthorized
     end
+  end
+  # rubocop:enable Metrics/MethodLength
+
+  def full_access?
+    users_permissions = load_users_permissions
+    return false unless users_permissions.is_a?(Array)
 
-    false
+    users_permissions.any? do |access|
+      access.permission == COMPLIANCE_FULL_ACCESS
+    end
   end
 
-  private
+  def verify(permitted, requested)
+    permitted_access = structurize(permitted)
+    requested_access = structurize(requested)
+    return false if permitted_access.app != requested_access.app
+    return false if permitted_access.resource != ANY &&
+                    permitted_access.resource != requested_access.resource
+
+    permitted_access.action == ANY ||
+      permitted_access.action == requested_access.action
+  end
```

Comment:
I thinki I understand what you're trying to do here, do you think it's possible to substitute it with [pattern matching](https://docs.ruby-lang.org/en/3.0/doc/syntax/pattern_matching_rdoc.html)? Maybe you'd have to use hashes instead of the structs though, but I guess it would require less jumps with guard cases.